src/main/java/org/cryptomator/ui/recoverykey/RecoveryKeyRecoverController.java

```diff
@@ -123,23 +142,61 @@ private boolean checkKeyAgainstVaultConfig(byte[] key) {
 
 	/* Getter/Setter */
 
+	public void validateRecoveryKey() {
+		var valid = recoveryKeyFactory.validateRecoveryKey(recoveryKey.get(), unverifiedVaultConfig != null ? this::checkKeyAgainstVaultConfig : null);
+		if (valid) {
+			recoveryKeyState.set(RecoveryKeyState.CORRECT);
+		} else {
+			if (recoveryKeyState.getValue() != RecoveryKeyState.WRONG) { //set via side effect in checkKeyAgainstVaultConfig
+				recoveryKeyState.set(RecoveryKeyState.INVALID);
+			}
+		}
+	}
```

Comment:
```suggestion
	public void validateRecoveryKey() {
		var valid = recoveryKeyFactory.validateRecoveryKey(recoveryKey.get(), unverifiedVaultConfig != null ? this::checkKeyAgainstVaultConfig : null);
		if (valid) {
			recoveryKeyState.set(RecoveryKeyState.CORRECT);
		} else if (recoveryKeyState.getValue() != RecoveryKeyState.WRONG) { // set via side effect in checkKeyAgainstVaultConfig()
			recoveryKeyState.set(RecoveryKeyState.INVALID);
		}
	}

	/* Getter/Setter */
```

Just a minor optimization in the if/else statement. But also moved the getter/setter comment down. I guess it's not really a getter/setter in the sense of JavaFX?